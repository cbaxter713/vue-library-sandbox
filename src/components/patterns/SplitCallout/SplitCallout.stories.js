import SplitCallout from "./SplitCallout.vue";

// More on default export: https://storybook.js.org/docs/vue/writing-stories/introduction#default-export
export default {
  title: "Patterns/Split Callout",
  component: SplitCallout,
  // More on argTypes: https://storybook.js.org/docs/vue/api/argtypes
  // argTypes: {
  //   size: {
  //     control: { type: "select" },
  //     options: ["sm", "md", "lg"],
  //   },
  //   theme: {
  //     control: { type: "select" },
  //     options: ["primary", "secondary", "action"],
  //   },
  // },
};

// More on component templates: https://storybook.js.org/docs/vue/writing-stories/introduction#using-args
const Template = (args) => ({
  // Components used in your story `template` are defined in the `components` object
  components: { SplitCallout },
  // The story's `args` need to be mapped into the template through the `setup()` method
  setup() {
    return { args };
  },
  // And then the `args` are bound to your component with `v-bind="args"`
  template: '<SplitCallout v-bind="args" />',
});

export const Default = Template.bind({});
Default.args = {
  title: "Default title",
  copy: "Lorem Ipsum and some other gibberish here",
  image: "https://via.placeholder.com/150",
};

export const ImageRight = Template.bind({});
ImageRight.args = {
  title: "Default title",
  copy: "Lorem Ipsum and some other gibberish here",
  image: "https://via.placeholder.com/150",
  rightAligned: true,
};
