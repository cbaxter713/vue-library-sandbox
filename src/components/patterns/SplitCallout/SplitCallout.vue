<script setup lang="ts">
export interface SplitCalloutProps {
  title: string;
  copy: string;
  image: string;
  rightAligned?: boolean;
}

const props = defineProps<SplitCalloutProps>();
</script>

<template>
  <div :class="['split-callout', { 'align-right': rightAligned }]">
    <div class="content">
      <h2>{{ title }}</h2>

      <p>{{ copy }}</p>
    </div>
    <img alt="some text" :src="image" />
  </div>
</template>

<style scoped lang="scss">
.split-callout {
  display: flex;
  gap: 2rem;
  width: 100%;
  max-width: 700px;
  margin: 0 auto;

  &.align-right {
    flex-direction: row-reverse;
  }
}
</style>
