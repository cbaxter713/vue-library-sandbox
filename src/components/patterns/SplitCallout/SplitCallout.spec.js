import { mount } from "@vue/test-utils";
import SplitCallout from "./SplitCallout.vue";

const wrapper = mount(SplitCallout, {
  props: {
    title: "Default title",
    copy: "Lorem Ipsum and some other gibberish here",
    image: "https://via.placeholder.com/150",
  },
});

describe("SplitCallout", () => {
  it("Should match the default snapshot", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
