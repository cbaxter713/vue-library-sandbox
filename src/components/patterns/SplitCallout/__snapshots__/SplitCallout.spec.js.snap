// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitCallout Should match the default snapshot 1`] = `
<div
  class="split-callout"
>
  <div
    class="content"
  >
    <h2>
      Default title
    </h2>
    <p>
      Lorem Ipsum and some other gibberish here
    </p>
  </div>
  <img
    alt="some text"
    src="https://via.placeholder.com/150"
  />
</div>
`;
